src/golang/cmd/server/handler/list_artifact_results.go

```diff
@@ -0,0 +1,155 @@
+	"github.com/aqueducthq/aqueduct/lib/workflow/artifact"
+	"github.com/dropbox/godropbox/errors"
+	"github.com/go-chi/chi/v5"
+	"github.com/google/uuid"
+)
+
+// Route: /workflow/{workflowId}/artifact/{artifactId}/results
+// Method: GET
+// Params: None
+// Request:
+//	Headers:
+//		`api-key`: user's API Key
+// Response:
+//	Body:
+//		serialized `listArtifactResultsResponse`
+
+type listArtifactResultsResponse struct {
+	Results []artifact.RawResultResponse `json:"results"`
+}
+
+type listArtifactResultsArgs struct {
+	*aq_context.AqContext
+	WorkflowId uuid.UUID
+	ArtifactId uuid.UUID
+}
+
+type ListArtifactResultsHandler struct {
+	GetHandler
+
+	Database             database.Database
+	ArtifactReader       db_artifact.Reader
+	ArtifactResultReader artifact_result.Reader
+	WorkflowDagReader    workflow_dag.Reader
+}
+
+func (*ListArtifactResultsHandler) Name() string {
+	return "ListArtifactResults"
+}
+
+func (*ListArtifactResultsHandler) Prepare(r *http.Request) (interface{}, int, error) {
+	aqContext, statusCode, err := aq_context.ParseAqContext(r.Context())
+	if err != nil {
+		return nil, statusCode, err
+	}
+
+	artfIdStr := chi.URLParam(r, routes.ArtifactIdUrlParam)
+	artfId, err := uuid.Parse(artfIdStr)
+	if err != nil {
+		return nil, http.StatusBadRequest, errors.Wrap(err, "Malformed artifact ID.")
+	}
+
+	wfIdStr := chi.URLParam(r, routes.WorkflowIdUrlParam)
+	wfId, err := uuid.Parse(wfIdStr)
+	if err != nil {
+		return nil, http.StatusBadRequest, errors.Wrap(err, "Malformed workflow Id.")
+	}
+
+	return &listArtifactResultsArgs{
+		AqContext:  aqContext,
+		ArtifactId: artfId,
+		WorkflowId: wfId,
+	}, http.StatusOK, nil
+}
+
+func (h *ListArtifactResultsHandler) Perform(ctx context.Context, interfaceArgs interface{}) (interface{}, int, error) {
+	args := interfaceArgs.(*listArtifactResultsArgs)
+	artfId := args.ArtifactId
+	wfId := args.WorkflowId
+
+	emptyResponse := listArtifactResultsResponse{}
+
+	artf, err := h.ArtifactReader.GetArtifact(ctx, artfId, h.Database)
+	if err != nil {
+		return emptyResponse, http.StatusInternalServerError, errors.Wrap(err, "Unable to retrieve artifact.")
+	}
+
+	results, err := h.ArtifactResultReader.GetArtifactResultsByArtifactNameAndWorkflowId(ctx, wfId, artf.Name, h.Database)
+	if err != nil {
+		return emptyResponse, http.StatusInternalServerError, errors.Wrap(err, "Unable to retrieve artifact results.")
+	}
+
+	if len(results) == 0 {
+		return emptyResponse, http.StatusOK, nil
+	}
+
+	resultIds := make([]uuid.UUID, 0, len(results))
+	for _, result := range results {
+		resultIds = append(resultIds, result.Id)
+	}
+
+	dbDagsByResultIds, err := h.WorkflowDagReader.GetWorkflowDagsMapByArtifactResultIds(ctx, resultIds, h.Database)
+	if err != nil {
+		return emptyResponse, http.StatusInternalServerError, errors.Wrap(err, "Unable to retrieve workflow dags.")
+	}
+
+	// maps from db dag Ids
+	dbDagsByDagIds := make(map[uuid.UUID]workflow_dag.DBWorkflowDag, len(dbDagsByResultIds))
```

Comment:
oh I meant like I usually use singular terminology for one-to-one mappings, so `dbDagByDagId`. But up to you, not a huge deal